Firestore/core/src/firebase/firestore/core/view_snapshot.mm

```diff
@@ -126,6 +129,79 @@
   return util::ToString(change_map_);
 }
 
+// ViewSnapshot
+
+ViewSnapshot::ViewSnapshot(FSTQuery* query,
+                           FSTDocumentSet* documents,
+                           FSTDocumentSet* old_documents,
+                           std::vector<DocumentViewChange> document_changes,
+                           model::DocumentKeySet mutated_keys,
+                           bool from_cache,
+                           bool sync_state_changed,
+                           bool excludes_metadata_changes)
+    : impl_{std::make_shared<const Impl>(query,
+                                         documents,
+                                         old_documents,
+                                         std::move(document_changes),
+                                         from_cache,
+                                         sync_state_changed,
+                                         excludes_metadata_changes,
+                                         std::move(mutated_keys))} {
+}
+
+ViewSnapshot ViewSnapshot::FromInitialDocuments(
+    FSTQuery* query,
+    FSTDocumentSet* documents,
+    DocumentKeySet mutated_keys,
+    bool from_cache,
+    bool excludes_metadata_changes) {
+  std::vector<DocumentViewChange> view_changes;
+  for (FSTDocument* doc in documents.documentEnumerator) {
+    view_changes.emplace_back(doc, DocumentViewChange::Type::kAdded);
+  }
+
+  return ViewSnapshot{
+      query, documents,
+      /*old_documents=*/
+      [FSTDocumentSet documentSetWithComparator:query.comparator],
+      std::move(view_changes), std::move(mutated_keys), from_cache,
+      /*sync_state_changed=*/true, excludes_metadata_changes};
+}
+
+std::string ViewSnapshot::ToString() const {
+  return StringFormat(
+      "<ViewSnapshot query: %s documents: %s old_documents: %s changes: %s "
+      "from_cache: %s mutated_keys: %s sync_state_changed: %s "
+      "excludes_metadata_changes: %s>",
+      query(), documents(), old_documents(),
+      objc::Description(document_changes()), from_cache(),
+      mutated_keys().size(), sync_state_changed(), excludes_metadata_changes());
+}
+
+size_t ViewSnapshot::Hash() const {
+  // Note: We are omitting `mutated_keys_` from the hash, since we don't have a
+  // straightforward way to compute its hash value.
```

Comment:
Is this a temporary problem or a permanent one? If temporary, consider rephrasing to "TODO: ..."

Reply:
The original comment had slightly more context, added the missing part (with very slight rewording) which I think makes things a little more clear.